lib/SIL/TypeLowering.cpp

```diff
@@ -1326,35 +1290,16 @@ const TypeLowering *TypeConverter::find(TypeKey k) {
   auto found = Types.find(ck);
   if (found == Types.end())
     return nullptr;
-  // We place a null placeholder in the hashtable to catch
```

Comment:
I think here you should have an assert that 'found->second' is not null, to catch recursion that wasn't handled in Sema.